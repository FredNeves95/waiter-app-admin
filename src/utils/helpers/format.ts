export const formatToBRCurrency = new Intl.NumberFormat('pt-BR', {
  style: 'currency',
  currency: 'BRL'
});
