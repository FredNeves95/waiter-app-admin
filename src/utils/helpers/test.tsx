import 'jsdom-global/register';
import { mount, shallow,   } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import { ReactNode } from 'react';
import { ActivePageProvider } from '../../contexts/ActivePageContext';
import { ModalContextProvider } from '../../contexts/ModalContext';
import { OrderContextProvider } from '../../contexts/OrderContext';

export function mountWithTheme(children: ReactNode) {
  return mount(
    <ThemeProvider theme={theme}>
      <ModalContextProvider>
        <ActivePageProvider>
          <OrderContextProvider>
            {children}
          </OrderContextProvider>
        </ActivePageProvider>
      </ModalContextProvider>
    </ThemeProvider>);
}

export function shallowWithTheme(children: ReactNode) {
  return shallow(
    <ThemeProvider theme={theme}>
      <ModalContextProvider>
        <ActivePageProvider>
          <OrderContextProvider>
            {children}
          </OrderContextProvider>
        </ActivePageProvider>
      </ModalContextProvider>
    </ThemeProvider>);
}
