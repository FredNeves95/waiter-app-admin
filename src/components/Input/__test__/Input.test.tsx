import { beforeEach, describe, expect, it, vi } from 'vitest';
import { Input, InputProps } from '../index';
import { mountWithTheme } from '../../../utils/helpers/test';


describe('<Input/>', () => {
  const handleChange = vi.fn();

  beforeEach(() => {
    vi.clearAllMocks();
  });

  const renderComponent = (props?: InputProps) =>
    mountWithTheme(<Input {...props} label='test' name='test' onChange={handleChange}/>);

  it('Should match the snapshot', () => {
    const input = renderComponent({
      placeholder: 'mocked placeholder',
      error: 'This field is required'
    });
    expect(input.html()).toMatchSnapshot();
  });

  it('Should render input', () => {
    const input = renderComponent();
    expect(input.isEmptyRender()).toBeFalsy();
  });

  it('Should render input with label received from props', () => {
    const input = renderComponent();
    expect(input.find('label').text()).toEqual('test');
  });

  it('Should render input with placeholder received from props', () => {
    const input = renderComponent({ placeholder: 'mocked placeholder' });
    expect(input.find('input').prop('placeholder')).toEqual('mocked placeholder');
  });

  it('Should execute handleChange function after typing in the input', () => {
    const input = renderComponent();

    const textInput = input.find('input');

    expect(handleChange).not.toHaveBeenCalled();

    textInput.simulate('change', { target: { name: 'test', value:'mocked typing' } });

    expect(handleChange).toHaveBeenCalled();
  });

  it('Should display error message when error is received from props', () => {
    const input = renderComponent({ error: 'This field is required' });
    expect(input.find('span').text()).toEqual('This field is required');
  });
});
