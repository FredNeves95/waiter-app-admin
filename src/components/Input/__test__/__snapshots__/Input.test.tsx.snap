// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Input/> > Should match the snapshot 1`] = `"<div class=\\"sc-bcPKhP lpbPOv\\"><label class=\\"sc-grXZZQ jlbCzr\\">test<input type=\\"text\\" placeholder=\\"mocked placeholder\\" name=\\"test\\" error=\\"This field is required\\" class=\\"sc-dkjKgF kUCshw\\"></label><span class=\\"sc-hLclGa ibFrep\\"><img src=\\"/src/assets/images/info.svg\\" alt=\\"Info\\">This field is required</span></div>"`;
